internal/stacks/stackruntime/internal/stackeval/provider_config.go

```diff
@@ -115,7 +116,15 @@ func (p *ProviderConfig) CheckProviderArgs(ctx context.Context) (cty.Value, tfdi
 			}
 			defer client.Close()
 
-			configVal, moreDiags := EvalBody(ctx, decl.Config, spec, ValidatePhase, p)
+			body := decl.Config
+			if body == nil {
+				// A provider with no configuration is valid (just means no
+				// attributes or blocks), but we need to pass an empty body to
+				// the evaluator to avoid a panic.
+				body = hcl.EmptyBody()
```

Comment:
Huh. Why wasn't this panicking before? Or was it?

Reply:
This function was never called previously, it would have panicked if it was.